src/portable/st/synopsys/dcd_synopsys.c

```diff
@@ -884,6 +892,24 @@ static void handle_rxflvl_ints(uint8_t rhport, USB_OTG_OUTEndpointTypeDef * out_
       TU_BREAKPOINT();
       break;
   }
+
+  // If an OUT EP was closed update (reduce) the RX FIFO size if RX FIFO is empty - since this function handle_rxflvl_ints() gets looped from dcd_int_handler() until RX FIFO is empty it is guaranteed to be entered
+  if (_rx_ep_closed && (usb_otg->GINTSTS & USB_OTG_GINTSTS_RXFLVL))
+  {
+    // Determine largest EP size for RX FIFO
+    uint16_t sz = xfer_status[0][TUSB_DIR_OUT].max_size;
+
+    for (uint8_t cnt = 1; cnt < EP_MAX; cnt++)
+    {
+      if (sz < xfer_status[cnt][TUSB_DIR_OUT].max_size) sz = xfer_status[cnt][TUSB_DIR_OUT].max_size;
+    }
+
+    // Update size of RX FIFO
+    usb_otg->GRXFSIZ = 15 + 2*sz/4 + 2*EP_MAX;        // sz was in bytes and is now needed in words
+
+    // Disable flag
+    _rx_ep_closed = false;
+  }
```

Comment:
we should move this handling code out the `handle_rxflvl_ints()` to https://github.com/hathach/tinyusb/blob/master/src/portable/st/synopsys/dcd_synopsys.c#L1050 Since we want to pull out all the packets in the fifo before changing its size. The `handle_rxflvl_ints()` function is calling within while loop fashion, we couldn't know for sure in the first pass all the data is pull out. 

Note: you could wrap it into its own function but it is optional.

Reply:
Very true, sorry i somehow thought the while loop is outside of dcd_synopsis ^^. I correct this!